ledger/complete/mtrie/trie/trie.go

```diff
@@ -352,16 +418,66 @@ func NewTrieWithUpdatedRegisters(
 	if err != nil {
 		return nil, 0, fmt.Errorf("constructing updated trie failed: %w", err)
 	}
+
+	// before returning the trie, add all payloads to storage
+	// TODO: avoid re-computing hash
+	// integrate into "update"
+	leafNodeUpdates := toLeafNodeUpdates(updatedPaths, updatedPayloads)
```

Comment:
This is an easier solution to find the updates to the storage. 

We could modify the `update` method to return the changes need to be added to storage, however, the change to update would be big. In order to keep the change simple and easier to review, I decided to re-compute the hash again here.